### PR TITLE
Works with CLI

### PR DESCRIPTION
**Pull request by 4rldur0**
#### 1) ```api.py```
- merge ```data.py``` into ```api.py```
- add ```extract_data``` function to get specific data such as titles, URLs, and status
- chage ```self.params``` to be private to protect the```api_key```
- exception handling: If the user enters ```limit``` less than 0, nothing is returned
#### 2) ```cli.py```
- add ```-n``` option: set the number of GIFs
- add ```-l``` option: show list of titles and URLs
- change ```-s``` option to ```-p``` option: print out all the response
#### 3) ```click_test.py```
- use ```CliRunner``` from ```click.testing```

### 2. Things to be added
- show GIFs in the terminal directly
- use commands without ```poetry```

### CODE REVIEW

File: unit_test.py [hunk 1]
```diff
@@ -1,38 +1,41 @@
...
 from api import GiphyAPI
-from data import Data
+
+gAPI = GiphyAPI()
 
 
 class Test(unittest.TestCase):
     # small tests
     def test_url(self):
-        api_key = "oZhufry2NyjOWfmCItORiADf8lPGTN2j"
```
Comment (semmypurewal) [inline unit_test.py:10]: Looks like your API key is now a part of your git history. That's like committing a password to your git repository. You will need to remove it. [Here are some instructions on how to do that](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository).

**Review by semmypurewal — COMMENTED** [review]
Looks good!

File: unit_test.py (hunk 1 above)
Comment (4rldur0) [reply unit_test.py:10]: That's what I deleted for this pull request! Now the API key doesn't appear in the file.

Comment (4rldur0) [reply unit_test.py:10]: That's what I deleted for this pull request! Now the API key doesn't appear in the file.

**Pull request merged**